xarray/core/common.py

```diff
@@ -133,15 +146,15 @@ def get_axis_num(self, dim):
         else:
             return tuple(self._get_axis_num(d) for d in dim)
 
-    def _get_axis_num(self, dim):
+    def _get_axis_num(self, dim: str) -> int:
         try:
             return self.dims.index(dim)
         except ValueError:
             raise ValueError("%r not found in array dimensions %r" %
                              (dim, self.dims))
 
     @property
-    def sizes(self):
+    def sizes(self) -> Frozen[str, int]:
```

Comment:
I don't think `Frozen` is a valid type annotation. Maybe better to stick with `Mapping[str, int]`?

Reply:
it is valid as I changed the Frozen class to be a subclass of typing.Mapping

Reply:
changed to ``Mapping[str, int]`` for the sake of user friendliness, as Frozen isn't really part of the public API